File: 프로그래머스/2/12923. 숫자 블록/숫자 블록.java
import java.util.*;

class Solution {
    public int[] solution(long begin, long end) {
        int[] answer = new int[(int) (end - begin) + 1];
        if (begin == 1) {
            for (int i = 1; i < answer.length; i ++) {
                answer[i] = 1;
            }
        } else {
            Arrays.fill(answer, 1);
        }
        
        for (long i = begin; i < end + 1; i++) {
            for (long n = 2; n < (int) Math.min(10000000, (int) Math.sqrt(i) + 1); n++) {
                if (i % n == 0) {
                    answer[(int)(i - begin)] = i / n <= 10000000 ? (int) Math.max(answer[(int)(i - begin)], i / n) : (int) Math.max(answer[(int)(i-begin)], n);
                }
            }
        }
        
        
        return answer;
    }
}

File: 프로그래머스/2/12923. 숫자 블록/숫자 블록.py
# 약수를 구한다
# 약수 중 자기 제외 가장 큰 수를 넣는다

def solution(begin, end):
    answer = [0] * (end - begin + 1)
    
    
    for n in range(begin, end + 1):
        for i in range(2, int(n**0.5) + 1):
            if n % i == 0:
                r = n // i
                if r <= 10_000_000:
                    answer[n-begin] = r
                    break
                else:
                    answer[n-begin] = i
                
                
        if answer[n-begin] == 0 and n > 1:
            answer[n-begin] = 1
    
    return answer
    